src/ontology/fbbt-edit.obo

```diff
@@ -127440,100 +127347,45 @@ is_obsolete: true
-name: synapsed_via_type_III_bouton_to
-def: "A relation between a motor neuron and a muscle that it synapses to via a type III bouton." [FBC:DOS]
-comment: Expands to: has_part some ('type III terminal button' that has_part some ('pre-synaptic membrane' that part_of some ('synapse' that has_part some ('post-synaptic membrane' that part_of some ?Y))))
 xref: RO:0002114
-is_a: synapsed_to ! synapsed_to
-inverse_of: synapsed_by_type_III_bouton_of ! synapsed_by_type_III_bouton_of
-expand_expression_to: "BFO_0000051 some (GO_0061177 that BFO_0000051 some (that BFO_0000051 some (GO_0097467 that BFO_0000051 some ( that BFO_0000050 some ?Y))))" []
 
 [Typedef]
 id: synapsed_via_type_II_bouton_to
-name: synapsed_via_type_II_bouton_to
-def: "A relation between a motor neuron and a muscle that it synapses to via a type II bouton." [FBC:DOS]
-comment: Expands to: has_part some ('type II terminal button' that has_part some ('pre-synaptic membrane' that part_of some ('synapse' that has_part some ('post-synaptic membrane' that part_of some ?Y))))
 xref: RO:0002107
-is_a: synapsed_to ! synapsed_to
-inverse_of: synapsed_by_type_II_bouton_of ! synapsed_by_type_II_bouton_of
-expand_expression_to: "BFO_0000051 some (GO_0061175 that BFO_0000051 some (that BFO_0000051 some (GO_0045202 that BFO_0000051 some ( that BFO_0000050 some ?Y))))" []
 
 [Typedef]
 id: synapsed_via_type_Ib_bouton_to
-name: synapsed_via_type_Ib_bouton_to
-def: "A relation between a motor neuron and a muscle that it synapses to via a type Ib bouton." [FBC:DOS]
-comment: Expands to: has_part some ('type Ib terminal button' that has_part some ('pre-synaptic membrane' that part_of some ('synapse' that has_part some ('post-synaptic membrane' that part_of some ?Y))))
 xref: RO:0002105
-is_a: synapsed_to ! synapsed_to
-inverse_of: synapsed_by_type_Ib_bouton_of ! synapsed_by_type_Ib_bouton_of
-expand_expression_to: "BFO_0000051 some (GO_0061176 that BFO_0000051 some (that BFO_0000051 some (GO_0045202 that BFO_0000051 some ( that BFO_0000050 some ?Y))))" []
 
 [Typedef]
 id: synapsed_via_type_Is_bouton_to
-name: synapsed_via_type_Is_bouton_to
-def: "A relation between a motor neuron and a muscle that it synapses to via a type Is bouton." [FBC:DOS]
-comment: Expands to: has_part some ('type Is terminal button' that has_part some ('pre-synaptic membrane' that part_of some ('synapse' that has_part some ('post-synaptic membrane' that part_of some ?Y))))
 xref: RO:0002106
-is_a: synapsed_to ! synapsed_to
-inverse_of: synapsed_by_type_Is_bouton_of ! synapsed_by_type_Is_bouton_of
-expand_expression_to: "BFO_0000051 some (GO_0061177 that BFO_0000051 some (that BFO_0000051 some (GO_0045202 that BFO_0000051 some ( that BFO_0000050 some ?Y))))" []
 
 [Typedef]
 id: tracheates
-name: tracheates
-def: "The relationships that holds between a trachea or tracheole and an anatomical structure that is contained in (and so provides an oxygen supply to)." [FBC:DOS]
 xref: RO:0002004
-is_a: contained_in ! contained_in
 
```

Comment:
@dosumis 

The file seems to process normally, but it can't be saved anymore. Unfortunately, I also do not understand the error messages that come with it. Could you try and see why it loads in protege, but cant be saved?

Reply:
Worked it out.  I think we may have reached the end of the line for using old OBO style relation IDs in the editors file.  This minimal OBO file is enough to replicate the issue:

```yaml
format-version: 1.2
date: 29:04:2018 20:25
import: http://purl.obolibrary.org/obo/fbbt/imports/ro_import.owl
ontology: fbbt

[Typedef]
id: part_of
xref: BFO:0000050
```

Try converting this to OWL and then back to obo:

```.sh
robot -vvv convert -i test.obo -f ofn -o test.owl
robot -vvv convert -i test.owl -f obo -o test_rt.obo
```

=> 
java.lang.NullPointerException
	at org.obolibrary.obo2owl.OWLAPIObo2Owl.translateShorthandIdToExpandedId(OWLAPIObo2Owl.java:1739)

I suspect one (or both?) of these axioms is causing the problem:

AnnotationAssertion(<http://www.geneontology.org/formats/oboInOwl#id> <http://purl.obolibrary.org/obo/BFO_0000050> "part_of"^^xsd:string)
AnnotationAssertion(<http://www.geneontology.org/formats/oboInOwl#shorthand> <http://purl.obolibrary.org/obo/BFO_0000050> "part_of"^^xsd:string)

This looks like a known bug (v.annoying misfeature of the obo2owl translator): OWL2OBO fails If the OBO ID value of the ID AP doesn't match the short_form of the iri.  This causes problems for classes.  It doesn't normally cause a problem for ObjectProperties, but looks like it does when he same OPs are imported.

This is not such a big deal for the editor's file, but I think we may need the old OBO IDs  for loading into FlyBase, given their current  pipeline- and this would be a big change to OBO release files if we have any legacy users who depend on them.  It *might* be possible to fix this by merging the axioms back in during production of the release file for FlyBase.  

CC  @cmungall

Reply:
In the meantime, what do I need to add back to the editors file to make it possible to save it again? The names?

Reply:
I thought we were going to have to remove the Typedefs completely (the point I was trying to make above), but on a whim I tested adding the name back and the roundtrip now works.  I tried a few more tests - it doesn't even seem to matter if ID and name match, as long as there's a name there!!  But no name + import of the same OP reliably => this roundtrip failure.  Weird.

Reply:
That did it! 1000x thanks!